model/src/main/java/com/plhal/ares/model/DataRepositoryImpl.java

```diff
@@ -186,36 +185,41 @@ private void zjistiSidlo() {
                 .concat(tempE.getElementsByTagName("dtt:Nazev_obce").item(0).getTextContent());
 
         // Nastavuji sídlo společnosti
-        firma.setSidlo(sidlo);
+        return sidlo;
 
     }
 
     // Získávám údaje o předmětech podnikání
-    private void pridejPredmetyPodnikani() {
+    private List<String> pridejPredmetyPodnikani(Document doc) {
+
+        List<String> predmetPodnikani = new ArrayList<>();
 
         // Získávám kolekci předmětů podnikání
-        tempNodeList = doc.getElementsByTagName("dtt:Predmet_podnikani");
-        tempE = (Element) tempNodeList.item(0);
+        NodeList tempNodeList = doc.getElementsByTagName("dtt:Predmet_podnikani");
+        Element tempE = (Element) tempNodeList.item(0);
 
         // Testuji, zda firma má předmět podnikání
         if (tempE == null) {
-            firma.getPredmetPodnikani().add("Neexistuje předmět podnikání");
-            return;
-        }
-        // Procházím kolekci předmětů podnikání
-        for (int i = 0; i < tempE.getElementsByTagName("dtt:Text").getLength(); i++) {
+            predmetPodnikani.add("Neexistuje předmět podnikání");
+        } else {
+            for (int i = 0; i < tempE.getElementsByTagName("dtt:Text").getLength(); i++) {
 
-            // Přidávám předměty podnikání do kolekce
-            firma.getPredmetPodnikani().add(tempE.getElementsByTagName("dtt:Text").item(i).getTextContent());
+                // Přidávám předměty podnikání do kolekce
+                predmetPodnikani.add(tempE.getElementsByTagName("dtt:Text").item(i).getTextContent());
+            }
         }
-
+        return predmetPodnikani;
     }
 
     // Získávám údaje o právní formě podnikání
-    private void zjistiPravniFormu() {
-        tempNodeList = doc.getElementsByTagName("dtt:Zakladni_udaje");
-        tempE = (Element) tempNodeList.item(0);
-        firma.setPravniForma(tempE.getElementsByTagName("dtt:Nazev_PF").item(0).getTextContent());
+    private String zjistiPravniFormu(Document doc) {
```

Comment:
Tady muzete take pouzit @NonNull. Na navratovem typu (zvazit, zda neni nullable - ale v navaznosti na datovou tridu si to nemyslim) i parametru.